### PR TITLE
plugin: allow to register static plugins for other packages

### LINKED ISSUE #43334: Make legacy plugin can be built into TiDB statically
## Enhancement

We have introduced a new implement of enterprise extension, see #42099 . Different with the old one, it will be built into TiDB statically instead of using go plugin mechanism. However, we should also maintain the old plugin for a while, and to avoid the burden of building go plugin, we can also move the legacy plugin build method to new one.

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #43334

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: plugin/plugin.go [hunk 1]
```diff
@@ -306,14 +308,58 @@ var testHook *struct {
...
+	}
+
+	p.plugins[pluginName] = plugin
+	return nil
+}
+
+// Get returns a registered plugin
+func (p *staticPlugins) Get(pluginName string) (m func() *Manifest, ok bool) {
+	p.Lock()
+	defer p.Unlock()
```
Comment (bb7133) [inline plugin/plugin.go:335]: You don't need to use `defer` here?

File: plugin/plugin.go [hunk 2]
```diff
@@ -306,14 +308,58 @@ var testHook *struct {
...
+	m, ok = p.plugins[pluginName]
+	return
+}
+
+// Clear clears all registered plugins
+func (p *staticPlugins) Clear() {
+	p.Lock()
+	defer p.Unlock()
```
Comment (bb7133) [inline plugin/plugin.go:343]: Ditto

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**